backend/kernelCI_app/helpers/issueExtras.py

```diff
@@ -43,40 +43,40 @@ def assign_issue_first_seen(
     """
     Assigns the first seen data to the processed_issues_table by querying with the issue_key_list.
     """
-    issue_id_list: List[str] = []
-    versions_per_issue: Dict[str, Set[int]] = defaultdict(set)
+    issue_id_list: set[str] = set()
+    versions_per_issue: dict[str, set[int]] = defaultdict(set)
+
     for issue_key in issue_key_list:
         issue_id, issue_version = issue_key
-        issue_id_list.append(issue_id)
+        issue_id_list.add(issue_id)
         versions_per_issue[issue_id].add(issue_version)
 
-    incident_records = get_issue_first_seen_data(issue_id_list=issue_id_list)
+    incident_records = get_issue_first_seen_data(issue_id_list=list(issue_id_list))
 
     for record in incident_records:
         record_issue_id = record["issue_id"]
         first_seen = record["first_seen"]
-        first_git_commit_hash = record["git_commit_hash"]
-        first_git_repository_url = record["git_repository_url"]
-        first_git_repository_branch = record["git_repository_branch"]
-        first_git_commit_name = record["git_commit_name"]
-        first_tree_name = record["tree_name"]
 
         processed_issue_from_id = processed_issues_table.get(record_issue_id)
         if processed_issue_from_id is None:
             processed_issues_table[record_issue_id] = {}
             processed_issue_from_id = processed_issues_table[record_issue_id]
             processed_issue_from_id["first_incident"] = FirstIncident(
                 first_seen=first_seen,
-                git_commit_hash=first_git_commit_hash,
-                git_repository_url=first_git_repository_url,
-                git_repository_branch=first_git_repository_branch,
-                git_commit_name=first_git_commit_name,
-                tree_name=first_tree_name,
+                git_commit_hash=record["git_commit_hash"],
+                git_repository_url=record["git_repository_url"],
+                git_repository_branch=record["git_repository_branch"],
+                git_commit_name=record["git_commit_name"],
+                tree_name=record["tree_name"],
+                issue_version=record["issue_version"],
             )
 
+        versions_table = processed_issue_from_id.get("versions")
+        if versions_table is None:
+            processed_issue_from_id["versions"] = {}
```

Comment:
wrt my other comment about

> just set "versions" above

```suggestion
        if processed_issue_from_id is None:
             first_incident = FirstIncident(
                first_seen = record["first_seen"]
                git_commit_hash=record["git_commit_hash"],
                git_repository_url=record["git_repository_url"],
                git_repository_branch=record["git_repository_branch"],
                git_commit_name=record["git_commit_name"],
                tree_name=record["tree_name"],
                issue_version=record["issue_version"],
            )
            # Create new processed issue and insert in the table
             processed_issues_table[record_issue_id] = processed_issue_from_id  = {
                 "versions": {},
                 "first_incident": first_incident
             }
```

Reply:
On top of this, we could even

```diff
-        processed_issue_from_id = processed_issues_table.get(record_issue_id)
-        if processed_issue_from_id is None:
+        processed_issue_from_id = processed_issues_table.setdefault(record_issue_id, {})
+        if not processed_issue_from_id:
             first_incident = FirstIncident(
	                first_seen = record["first_seen"]
	                git_commit_hash=record["git_commit_hash"],
	                git_repository_url=record["git_repository_url"],
	                git_repository_branch=record["git_repository_branch"],
	                git_commit_name=record["git_commit_name"],
	                tree_name=record["tree_name"],
	                issue_version=record["issue_version"],
	            )
	            # Fill new processed issue
	             processed_issue_from_id | = {
	                 "versions": {},
	                 "first_incident": first_incident
	             }
```

Reply:
Yeah, there are some built-ins that we only learn from other people, unless we take a day to read the docs, and ain't nobody got time for that 😆